### ISSUE TITLE
aws_ssm: StringList SSM Parameter cannot contain the ',' character

### ISSUE THREAD

**Issue by jshinevar** [aws/aws-cdk#26944]
### Describe the bug

When creating a StringList SSM Parameter, it will not allow a list of strings separated by commas.  This is definitely allowed because you can manually type in a list inside of SSM in AWS.

### Expected Behavior

I would expect the `string_list_value` parameter to allow a comma in it's contents.

### Current Behavior

on `cdk synth` I get the following error: `RuntimeError: Values of a StringList SSM Parameter cannot contain the ',' character. Use a string parameter instead.`

### Reproduction Steps

create a aws_ssm.StringListParameter and in the `string_list_value` parameter have a comma in the string

### CDK CLI Version

2.93.0

### Node.js Version

9.5.1

### OS

MacOS and Linux

### Language

Python

### Language Version

3.9.6

**Comment by peterwoodworth** [comment]
According to the API docs, [commas are not allowed values](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_Tag.html#systemsmanager-Type-Tag-Value). Could you please give an example of the list you are attempting to create and the CDK code that you are doing that with?

**Comment by jshinevar** [comment]
<img width="1404" alt="image" src="https://github.com/aws/aws-cdk/assets/16395879/f0416247-9dea-466a-a50f-52667d686ae3">

**Comment by jshinevar** [comment]
<img width="1404" alt="image" src="https://github.com/aws/aws-cdk/assets/16395879/cdc95e0b-5e1c-43a7-8c19-abe3414f4332">

@peterwoodworth please see the image above

**Comment by peterwoodworth** [comment]
Right, just do this

```
    new ssm.StringListParameter(this, 'stringlistparam', {
      stringListValue: ['secret1', 'secret2', 'secret3']
    })
```

**Comment by jshinevar** [comment]
The data being provided is being read out of a text file, I mean, I guess I can read it, remove the commas and create an array and then put it back in, but that seems like a lot of unnecessary logic when CDK should be able to just differentiate when presented.  My temporary solution is going to be just seeing if a String Parameter works vice a StringList... but it seems to me that if you are having a "StringList" you should either be able to accept an array of strings or a string with the expected delimeter.

**Comment by akmalharith** [comment]
`stringListValue` is expected to be `string[]` 

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ssm.StringListParameter.html.

**Comment by peterwoodworth** [comment]
Yes, the high level construct is designed this way. If it creates too much of a hassle for you, you could provide some dummy value to the `stringListValue` prop, and then use an escape hatch to override the synthesized `value` property on the `defaultChild` of the `StringListParameter` with your comma separated list

**Comment by pradoz** [comment]
For anyone who lands here in the future, StringListParameter has been deprecated. I found the easiest here is either
1. Store the strings with a well-known delimiter and splitting/parsing it based on that delimiter
2. Passing stringify'd JSON to the parameter if a map is needed, then parsing that value into a JSON structure and performing key lookups on the resulting map
